hraven-etl/src/main/java/com/twitter/hraven/mapreduce/GraphiteHistoryWriter.java

```diff
@@ -372,6 +376,62 @@ public void write() throws IOException {
     socketWriter.write(lines.toString());
   }
 
+    private int getTimeStamp() {
+        int timestamp;
+
+//        timestamp = Math.round(recordCollection.getSubmitTime() / 1000);
+//        return timestamp;
+
+
+        StandardEvaluationContext context = new StandardEvaluationContext();
+
+        try {
+            context.registerFunction("dateToEpoch", GraphiteHistoryWriter.class.getDeclaredMethod("dateToEpoch", String.class));
+            context.registerFunction("submitTime", GraphiteHistoryWriter.class.getDeclaredMethod("submitTime",JobHistoryRecordCollection.class));
+        } catch (SecurityException e) {
+            LOG.error("SecurityException while adding methods to SEPL context", e);
+            throw new RuntimeException(e);
+        } catch (NoSuchMethodException e) {
+            LOG.error("NoSuchMethodException while adding methods to SEPL context", e);
+            throw new RuntimeException(e);
+        }
+        context.setVariable("tag", userTokens);
+        context.setVariable("records", recordCollection);
+
+        Expression filterExp = getParsedExpression(sinkConfig.getTimestampFilter(), false);
+        LOG.info("filterExp: " + filterExp.getExpressionString() );
+        LOG.info("userTokens: " + userTokens.toString());
+        Long timestampLong = filterExp.getValue(context, Long.class);
+        if(timestampLong != null) {
+            timestamp = Math.round(timestampLong / 1000);
```

Comment:
don't need this extra variable timestamp.. simply return Math.round (timestamp--Long--/1000) if it is not null else Math.round(submitTime(..)/1000)